examples/http-storage/index.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016-2019, Pulumi Corporation.  All rights reserved.
+
+import * as azure from '@pulumi/azure';
+import { HttpRequest } from '@pulumi/azure/appservice';
+
+const resourceGroup = new azure.core.ResourceGroup('example', { location: 'West US 2' });
+
+const storageAccount = new azure.storage.Account("storage", {
+  resourceGroupName: resourceGroup.name,
+  accountReplicationType: "LRS",
+  accountTier: "Standard",
+});
+
+const args: azure.appservice.HttpEventSubscriptionArgs = {
+  resourceGroup,
+  callback: async (context, req: HttpRequest) => {
+
+    context.bindings.person = [];
+    context.bindings.person.push({
+      PartitionKey: req.query.email,
+      RowKey: req.query.email,
+      Name: req.query.name
+    });
```

Comment:
So, the [recommended](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node) way to use input and output bindings is to accept inputs as arguments and return outputs as properties:

``` ts
async function(context, myTrigger, myInput, myOtherInput) {
    let retMsg = `Hello, ${myInput} and ${myOtherInput}!`;
    return {
        httpResponse: {
            body: retMsg
        },
        queueOutput: retMsg
    };
}
```

I would very much like to encourage this style of callbacks and to support it as the default option. Ideally, we would find a TypeScript sorcery to check the number of arguments/returns versus the names and quantities of the bindings, but we may also do this at preview execution time.

Reply:
This comment basically applies only to the example, not the pulumi sdk part of the PR, right? I can change that, I just used a code sample that was on the docs (https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-table#output---javascript-example). I can fix this, though.

Reply:
I suppose the current SDK won't let you change your example to mine: types define the callback signature quite strictly.
What's your opinion about which style of binding assignment is preferable?